tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/ExerciseBuilder.java

```diff
@@ -0,0 +1,122 @@
+package fi.helsinki.cs.tmc.langs;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+
+public class ExerciseBuilder {
+
+    private String beginSolution = "// BEGIN SOLUTION";
+    private String endSolution = "// END SOLUTION";
+    private String stubMarker = "// STUB:";
+    private String sourceFolderName = "src";
+
+    /**
+     * Constuctor
+     *
+     * <p>
+     * TODO: Needs to be expanded with LanguagePlugin, which determines the
+     * comment style used with the language
```

Comment:
The class hierarchy in `tmc-server/lib/course_refresher` might be useful to look at.

I don't think this class needs to know about language plugins but language plugins can use this class (or some subclass) as a helper.